modules/router-store/src/router_store_module.ts

```diff
@@ -377,28 +389,31 @@ export class StoreRouterConnectingModule {
   private dispatchRouterNavigation(
     lastRoutesRecognized: RoutesRecognized
   ): void {
+    const nextRouterState = this.serializer.serialize(
+      lastRoutesRecognized.state
+    );
     this.dispatchRouterAction(ROUTER_NAVIGATION, {
-      routerState: this.routerState,
+      routerState: nextRouterState,
       event: new RoutesRecognized(
         lastRoutesRecognized.id,
         lastRoutesRecognized.url,
         lastRoutesRecognized.urlAfterRedirects,
-        this.routerState
+        nextRouterState
       ),
     });
   }
 
   private dispatchRouterCancel(event: NavigationCancel): void {
     this.dispatchRouterAction(ROUTER_CANCEL, {
-      routerState: this.routerState,
+      routerState: this.routerState!,
```

Comment:
Is there a reason why `this.routerState!` is used?
Is it to make clear the router state can't be null at this point?

Reply:
Yes, that's the only reason.